src/libs/history.c

```diff
@@ -193,10 +212,29 @@ static GList *_duplicate_history(GList *hist)
 
     memcpy(new, old, sizeof(dt_dev_history_item_t));
 
-    new->params = malloc(old->module->params_size);
+    int32_t params_size = 0;
+    if(old->module)
+    {
+      params_size = old->module->params_size;
+    }
+    else
+    {
+      dt_iop_module_t *base = get_base_module(darktable.develop, old->op_name);
+      if(base)
+      {
+        params_size = base->params_size;
+      }
+      else
+      {
+        // nothing else to do
+        fprintf(stderr, "[_duplicate_history] can't find base module for %s\n", old->op_name);
```

Comment:
Why the underscore just after [? Is that a dt style, I never paid attention to that.

Reply:
Is the name of the function.
No idea if there's a standard for function naming, woudn't hurt to have one. As far as the printf, if this isn't a standard for errors that just can't happen it should be, it make it easy to see where it comes from when reported.